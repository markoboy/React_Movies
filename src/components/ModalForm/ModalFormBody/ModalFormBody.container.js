import {
  MODAL_ADD_TITLE,
  MODAL_DELETE_TITLE,
  MODAL_EDIT_TITLE,
  MODAL_FORM_ADD_ACTION,
  MODAL_FORM_DELETE_ACTION,
  MODAL_FORM_EDIT_ACTION,
  MODAL_FORM_EDIT_PARAGRAPH,
} from '@constants/Modal';
import FormInputs from '@components/ModalForm/FormInputs';
import { setModalTitleCreator } from '@store/action-creators/modalActionCreators';
import { setFormInputsCreator } from '@store/action-creators/modalFormActionCreators';
import {
  modalFormActionSelector,
  modalFormInputsSelector,
  modalFormSelectedMovieSelector,
} from '@store/selectors/modalFormSelectors';
import getModalFormInputs from '@utils/getModalFormInputs';
import React, { memo, useEffect, useMemo } from 'react';
import { connect } from 'react-redux';

function ModalFormBodyContainer({ action, selectedMovie, dispatch }) {
  const formInputs = useMemo(() => getModalFormInputs(selectedMovie), [selectedMovie]);

  const [hasDeleteBody, hasFormBody] = useMemo(() => {
    const deleteBody = action === MODAL_FORM_DELETE_ACTION;
    const formBody = [MODAL_FORM_ADD_ACTION, MODAL_FORM_EDIT_ACTION].includes(action);

    return [deleteBody, formBody];
  }, [action]);

  useEffect(() => {
    switch (action) {
      case MODAL_FORM_ADD_ACTION:
        dispatch(setModalTitleCreator(MODAL_ADD_TITLE));
        dispatch(setFormInputsCreator(formInputs));
        break;

      case MODAL_FORM_EDIT_ACTION:
        dispatch(setModalTitleCreator(MODAL_EDIT_TITLE));
        dispatch(setFormInputsCreator(formInputs));
        break;

      case MODAL_FORM_DELETE_ACTION:
        dispatch(setModalTitleCreator(MODAL_DELETE_TITLE));
        break;

      default:
        break;
    }
  }, [action]);

  return (
    <>
      {hasFormBody && <FormInputs />}
      {hasDeleteBody && (
        <p>{MODAL_FORM_EDIT_PARAGRAPH}</p>
      )}
    </>
  );
}

const mapStateToProps = (state) => ({
  action: modalFormActionSelector(state),
  selectedMovie: modalFormSelectedMovieSelector(state),
  formInputs: modalFormInputsSelector(state),
});

export default connect(mapStateToProps)(memo(ModalFormBodyContainer));
