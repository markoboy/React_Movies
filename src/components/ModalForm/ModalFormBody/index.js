export { default } from './ModalFormBody.container';
