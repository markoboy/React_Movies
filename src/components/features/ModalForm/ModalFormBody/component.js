import PropTypes from 'prop-types';
import React from 'react';
import { MODAL_FORM_DELETE_PARAGRAPH } from '../constants';
import FormikForm from '../FormikForm';
import ModalFormFooter from '../ModalFormFooter';

function ModalFormBodyComponent({
  hasFormBody,
  hasDeleteBody,
  selectedMovie,
  formAction,
  onSubmit,
}) {
  return (
    <>
      {hasFormBody && (
        <FormikForm selectedMovie={selectedMovie} onSubmit={onSubmit}>
          <ModalFormFooter action={formAction} />
        </FormikForm>
      )}

      {hasDeleteBody && (
        <form onSubmit={onSubmit}>
          <p>{MODAL_FORM_DELETE_PARAGRAPH}</p>
          <ModalFormFooter action={formAction} />
        </form>
      )}
    </>
  );
}

ModalFormBodyComponent.defaultProps = {
  selectedMovie: undefined,
};

ModalFormBodyComponent.propTypes = {
  hasFormBody: PropTypes.bool.isRequired,
  hasDeleteBody: PropTypes.bool.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  selectedMovie: PropTypes.object,
  formAction: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default ModalFormBodyComponent;
