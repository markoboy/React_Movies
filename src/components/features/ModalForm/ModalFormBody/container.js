import {
  MODAL_ADD_TITLE,
  MODAL_DELETE_TITLE,
  MODAL_EDIT_TITLE,
  MODAL_FORM_ADD_ACTION,
  MODAL_FORM_DELETE_ACTION,
  MODAL_FORM_EDIT_ACTION,
} from '@components/features/ModalForm/constants';
import { MovieDetailType } from '@constants/MovieTypes';
import PropTypes from 'prop-types';
import React, { useEffect, useMemo } from 'react';
import { getModalFormInputs } from '../utils';
import ModalFormBodyComponent from './component';

export default function ModalFormBodyContainer({
  action,
  selectedMovie,
  setModalTitle,
  setFormInputs,
}) {
  const formInputs = useMemo(() => getModalFormInputs(selectedMovie), [
    selectedMovie,
  ]);

  const [hasDeleteBody, hasFormBody] = useMemo(() => {
    const deleteBody = action === MODAL_FORM_DELETE_ACTION;
    const formBody = [MODAL_FORM_ADD_ACTION, MODAL_FORM_EDIT_ACTION].includes(
      action
    );

    return [deleteBody, formBody];
  }, [action]);

  useEffect(() => {
    switch (action) {
      case MODAL_FORM_ADD_ACTION:
        setModalTitle(MODAL_ADD_TITLE);
        setFormInputs(formInputs);
        break;

      case MODAL_FORM_EDIT_ACTION:
        setModalTitle(MODAL_EDIT_TITLE);
        setFormInputs(formInputs);
        break;

      case MODAL_FORM_DELETE_ACTION:
        setModalTitle(MODAL_DELETE_TITLE);
        break;

      default:
        break;
    }
  }, [action]);

  return (
    <ModalFormBodyComponent
      hasFormBody={hasFormBody}
      hasDeleteBody={hasDeleteBody}
    />
  );
}

ModalFormBodyContainer.defaultProps = {
  selectedMovie: null,
};

ModalFormBodyContainer.propTypes = {
  action: PropTypes.string.isRequired,
  selectedMovie: PropTypes.shape(MovieDetailType),
  setModalTitle: PropTypes.func.isRequired,
  setFormInputs: PropTypes.func.isRequired,
};
