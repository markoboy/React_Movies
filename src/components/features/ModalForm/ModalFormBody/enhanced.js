import { setModalTitleCreator } from '@store/action-creators/modalActionCreators';
import { setFormInputsCreator } from '@store/action-creators/modalFormActionCreators';
import {
  modalFormActionSelector,
  modalFormInputsSelector,
  modalFormSelectedMovieSelector,
} from '@store/selectors/modalFormSelectors';
import { memo } from 'react';
import { connect } from 'react-redux';
import ModalFormBodyContainer from './container';

const mapStateToProps = (state) => ({
  action: modalFormActionSelector(state),
  selectedMovie: modalFormSelectedMovieSelector(state),
  formInputs: modalFormInputsSelector(state),
});

const mapDispatchToProps = {
  setModalTitle: setModalTitleCreator,
  setFormInputs: setFormInputsCreator,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(memo(ModalFormBodyContainer));
