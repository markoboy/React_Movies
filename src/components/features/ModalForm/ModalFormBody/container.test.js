import React from 'react';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import ModalFormBody from './container';
import {
  MODAL_ADD_TITLE,
  MODAL_DELETE_TITLE,
  MODAL_EDIT_TITLE,
  MODAL_FORM_ADD_ACTION,
  MODAL_FORM_DELETE_ACTION,
  MODAL_FORM_EDIT_ACTION,
} from '../constants';

describe('ModalFormBody Component', () => {
  let setModalTitle;
  let onSubmit;

  beforeEach(() => {
    setModalTitle = jest.fn();
    onSubmit = jest.fn();
  });

  it('renders a form with inputs when action is Add', () => {
    const { container } = render(
      <ModalFormBody
        action={MODAL_FORM_ADD_ACTION}
        selectedMovie={null}
        setModalTitle={setModalTitle}
        onSubmit={onSubmit}
      />
    );

    expect(setModalTitle).toHaveBeenCalledTimes(1);
    expect(setModalTitle).toHaveBeenCalledWith(MODAL_ADD_TITLE);

    expect(container).toMatchSnapshot();
  });

  it('renders a form with inputs when action is Edit', () => {
    const { container } = render(
      <ModalFormBody
        action={MODAL_FORM_EDIT_ACTION}
        selectedMovie={null}
        setModalTitle={setModalTitle}
        onSubmit={onSubmit}
      />
    );

    expect(setModalTitle).toHaveBeenCalledTimes(1);
    expect(setModalTitle).toHaveBeenCalledWith(MODAL_EDIT_TITLE);

    expect(container).toMatchSnapshot();
  });

  it('renders a form with delete paragraph when action is Delete', () => {
    const { container } = render(
      <ModalFormBody
        action={MODAL_FORM_DELETE_ACTION}
        selectedMovie={null}
        setModalTitle={setModalTitle}
        onSubmit={onSubmit}
      />
    );

    expect(setModalTitle).toHaveBeenCalledTimes(1);
    expect(setModalTitle).toHaveBeenCalledWith(MODAL_DELETE_TITLE);

    expect(container).toMatchSnapshot();
  });

  it('triggers onSubmit with values and formik instance when action is Add', async () => {
    const { getByText, getByLabelText, getByRole } = render(
      <ModalFormBody
        action={MODAL_FORM_ADD_ACTION}
        selectedMovie={null}
        setModalTitle={setModalTitle}
        onSubmit={onSubmit}
      />
    );

    userEvent.type(getByLabelText(/title/i), 'A film title');
    userEvent.type(
      getByLabelText(/Movie Poster URL/i),
      'https://film.com/image.jpg'
    );
    userEvent.type(getByLabelText(/Overview/i), 'A great film');
    userEvent.type(getByLabelText(/Runtime/i), '145');

    const selectElement = getByText(/Select Genres/i);

    userEvent.click(selectElement);

    const documentary = getByText(/Documentary/i);
    userEvent.click(documentary);

    userEvent.click(getByRole('button', { name: /submit/i }));

    await waitFor(() => {
      expect(onSubmit).toHaveBeenCalledTimes(1);
      expect(onSubmit).toHaveBeenCalledWith(
        expect.objectContaining({
          title: 'A film title',
          poster_path: 'https://film.com/image.jpg',
          overview: 'A great film',
          runtime: 145,
          genres: [{ label: 'Documentary', value: 'documentary' }],
        }),
        expect.anything(),
      );
    });
  });

  it('renders form with selectedMovie values when action is MODAL_FORM_EDIT_ACTION and form is reset', async () => {
    const selectedMovie = {
      id: 1,
      title: 'A film title',
      poster_path: 'https://film.com/image.jpg',
      overview: 'A great film',
      runtime: 145,
      genres: ['Documentary'],
    };

    const { container } = render(
      <ModalFormBody
        action={MODAL_FORM_EDIT_ACTION}
        selectedMovie={selectedMovie}
        setModalTitle={setModalTitle}
        onSubmit={onSubmit}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
