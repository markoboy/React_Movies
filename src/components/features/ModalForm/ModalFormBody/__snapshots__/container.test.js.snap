// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFormBody Component renders a form with delete paragraph when action is Delete 1`] = `
<div>
  <form>
    <p>
      Are you sure you want to delete this movie?
    </p>
    <div
      class="row justify-content--flex-end"
    >
      <div
        class="column column--m-3"
      >
        <button
          class="btn btn--primary btn--full-width"
          tabindex="0"
          type="submit"
        >
          Confirm
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`ModalFormBody Component renders a form with inputs when action is Add 1`] = `
<div>
  <form>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="title"
      >
        Title
      </label>
      <input
        class="form__input"
        id="title"
        name="title"
        placeholder="Title here"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for=""
      >
        Release Date
      </label>
      <div
        class="react-date-picker react-date-picker--closed react-date-picker--enabled"
      >
        <div
          class="react-date-picker__wrapper"
        >
          <div
            class="react-date-picker__inputGroup"
          >
            <input
              max="275760-09-13"
              min="0001-01-01"
              name="date"
              style="visibility: hidden; position: absolute; z-index: -999;"
              type="date"
              value=""
            />
            
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
              data-input="true"
              inputmode="numeric"
              max="31"
              min="1"
              name="day"
              placeholder="Day"
              style="width: 17px;"
              type="number"
              value=""
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              -
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
              data-input="true"
              inputmode="numeric"
              max="12"
              min="1"
              name="month"
              placeholder="Month"
              style="width: 28px;"
              type="number"
              value=""
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              -
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
              data-input="true"
              inputmode="numeric"
              max="275760"
              min="1"
              name="year"
              placeholder="Year"
              step="1"
              style="width: 20px;"
              type="number"
              value=""
            />
            
          </div>
          <button
            class="react-date-picker__calendar-button react-date-picker__button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-calendar-alt fa-w-14 "
              data-icon="calendar-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="poster_path"
      >
        Movie Poster URL
      </label>
      <input
        class="form__input"
        id="poster_path"
        name="poster_path"
        placeholder="Movie Poster URL here"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for=""
      >
        Genre
      </label>
      <div
        class="go2139093995 multi-select"
      >
        <div
          aria-readonly="true"
          class="go2646822163 dropdown-container"
          tabindex="0"
        >
          <div
            class="go2642161244 dropdown-heading"
          >
            <div
              class="dropdown-heading-value"
            >
              <span
                class="gray"
              >
                Select Genres
              </span>
            </div>
            <svg
              class="dropdown-heading-dropdown-arrow gray"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-width="2"
              width="24"
            >
              <path
                d="M6 9L12 15 18 9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="overview"
      >
        Overview
      </label>
      <input
        class="form__input"
        id="overview"
        name="overview"
        placeholder="Overview here"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="runtime"
      >
        Runtime
      </label>
      <input
        class="form__input"
        id="runtime"
        name="runtime"
        placeholder="Runtime here"
        type="number"
        value=""
      />
    </div>
    <div
      class="row justify-content--flex-end"
    >
      <div
        class="column column--m-3"
      >
        <button
          class="btn btn--outline btn--full-width"
          tabindex="0"
          type="reset"
        >
          Reset
        </button>
      </div>
      <div
        class="column column--m-3"
      >
        <button
          class="btn btn--primary btn--full-width"
          tabindex="0"
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`ModalFormBody Component renders a form with inputs when action is Edit 1`] = `
<div>
  <form>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="title"
      >
        Title
      </label>
      <input
        class="form__input"
        id="title"
        name="title"
        placeholder="Title here"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for=""
      >
        Release Date
      </label>
      <div
        class="react-date-picker react-date-picker--closed react-date-picker--enabled"
      >
        <div
          class="react-date-picker__wrapper"
        >
          <div
            class="react-date-picker__inputGroup"
          >
            <input
              max="275760-09-13"
              min="0001-01-01"
              name="date"
              style="visibility: hidden; position: absolute; z-index: -999;"
              type="date"
              value=""
            />
            
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
              data-input="true"
              inputmode="numeric"
              max="31"
              min="1"
              name="day"
              placeholder="Day"
              style="width: 17px;"
              type="number"
              value=""
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              -
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
              data-input="true"
              inputmode="numeric"
              max="12"
              min="1"
              name="month"
              placeholder="Month"
              style="width: 28px;"
              type="number"
              value=""
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              -
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
              data-input="true"
              inputmode="numeric"
              max="275760"
              min="1"
              name="year"
              placeholder="Year"
              step="1"
              style="width: 20px;"
              type="number"
              value=""
            />
            
          </div>
          <button
            class="react-date-picker__calendar-button react-date-picker__button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-calendar-alt fa-w-14 "
              data-icon="calendar-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="poster_path"
      >
        Movie Poster URL
      </label>
      <input
        class="form__input"
        id="poster_path"
        name="poster_path"
        placeholder="Movie Poster URL here"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for=""
      >
        Genre
      </label>
      <div
        class="go2139093995 multi-select"
      >
        <div
          aria-readonly="true"
          class="go2646822163 dropdown-container"
          tabindex="0"
        >
          <div
            class="go2642161244 dropdown-heading"
          >
            <div
              class="dropdown-heading-value"
            >
              <span
                class="gray"
              >
                Select Genres
              </span>
            </div>
            <svg
              class="dropdown-heading-dropdown-arrow gray"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-width="2"
              width="24"
            >
              <path
                d="M6 9L12 15 18 9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="overview"
      >
        Overview
      </label>
      <input
        class="form__input"
        id="overview"
        name="overview"
        placeholder="Overview here"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="runtime"
      >
        Runtime
      </label>
      <input
        class="form__input"
        id="runtime"
        name="runtime"
        placeholder="Runtime here"
        type="number"
        value=""
      />
    </div>
    <div
      class="row justify-content--flex-end"
    >
      <div
        class="column column--m-3"
      >
        <button
          class="btn btn--outline btn--full-width"
          tabindex="0"
          type="reset"
        >
          Reset
        </button>
      </div>
      <div
        class="column column--m-3"
      >
        <button
          class="btn btn--primary btn--full-width"
          tabindex="0"
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`ModalFormBody Component renders form with selectedMovie values when action is MODAL_FORM_EDIT_ACTION and form is reset 1`] = `
<div>
  <form>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="id"
      >
        Movie ID
      </label>
      <input
        class="form__input"
        disabled=""
        id="id"
        name="id"
        placeholder=""
        type="number"
        value="1"
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="title"
      >
        Title
      </label>
      <input
        class="form__input"
        id="title"
        name="title"
        placeholder="Title here"
        type="text"
        value="A film title"
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for=""
      >
        Release Date
      </label>
      <div
        class="react-date-picker react-date-picker--closed react-date-picker--enabled"
      >
        <div
          class="react-date-picker__wrapper"
        >
          <div
            class="react-date-picker__inputGroup"
          >
            <input
              max="275760-09-13"
              min="0001-01-01"
              name="date"
              style="visibility: hidden; position: absolute; z-index: -999;"
              type="date"
              value=""
            />
            
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
              data-input="true"
              inputmode="numeric"
              max="31"
              min="1"
              name="day"
              placeholder="Day"
              style="width: 17px;"
              type="number"
              value=""
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              -
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
              data-input="true"
              inputmode="numeric"
              max="12"
              min="1"
              name="month"
              placeholder="Month"
              style="width: 28px;"
              type="number"
              value=""
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              -
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
              data-input="true"
              inputmode="numeric"
              max="275760"
              min="1"
              name="year"
              placeholder="Year"
              step="1"
              style="width: 20px;"
              type="number"
              value=""
            />
            
          </div>
          <button
            class="react-date-picker__calendar-button react-date-picker__button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-calendar-alt fa-w-14 "
              data-icon="calendar-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="poster_path"
      >
        Movie Poster URL
      </label>
      <input
        class="form__input"
        id="poster_path"
        name="poster_path"
        placeholder="Movie Poster URL here"
        type="text"
        value="https://film.com/image.jpg"
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for=""
      >
        Genre
      </label>
      <div
        class="go2139093995 multi-select"
      >
        <div
          aria-readonly="true"
          class="go2646822163 dropdown-container"
          tabindex="0"
        >
          <div
            class="go2642161244 dropdown-heading"
          >
            <div
              class="dropdown-heading-value"
            >
              <span>
                Documentary
              </span>
            </div>
            <button
              aria-label="Clear Selected"
              class="go1003880112 clear-selected-button"
              type="button"
            >
              <svg
                class="dropdown-search-clear-icon gray"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-width="2"
                width="24"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </button>
            <svg
              class="dropdown-heading-dropdown-arrow gray"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-width="2"
              width="24"
            >
              <path
                d="M6 9L12 15 18 9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="overview"
      >
        Overview
      </label>
      <input
        class="form__input"
        id="overview"
        name="overview"
        placeholder="Overview here"
        type="text"
        value="A great film"
      />
    </div>
    <div
      class="form-element-wrapper"
    >
      <label
        class="form__label"
        for="runtime"
      >
        Runtime
      </label>
      <input
        class="form__input"
        id="runtime"
        name="runtime"
        placeholder="Runtime here"
        type="number"
        value="145"
      />
    </div>
    <div
      class="row justify-content--flex-end"
    >
      <div
        class="column column--m-3"
      >
        <button
          class="btn btn--outline btn--full-width"
          tabindex="0"
          type="reset"
        >
          Reset
        </button>
      </div>
      <div
        class="column column--m-3"
      >
        <button
          class="btn btn--primary btn--full-width"
          tabindex="0"
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;
